crates/formatter/src/format_tokens_macro.rs

```diff
@@ -0,0 +1,61 @@
+/// The macro `format_tokens` is a convenience macro to
+/// use when writing a list of tokens that should be at the same level
+/// without particular rule.
+///
+/// # Examples
+///
+/// Let's suppose you need to create tokens for the string `"foo": "bar"`,
+/// you would write:
+///
+/// ```rust
+/// use rome_formatter::{format_tokens, FormatToken};
+/// let token = format_tokens!("foo:", FormatToken::Space, "bar");
+/// ```
+///
+/// The macro can be also nested, although the macro needs to be decorated with the token you need.
+/// For example, let's try to format following string:
+///
+/// ```no_rust
+/// foo: { bar: lorem }
+/// ```
+/// You would write it like the following:
+///
+/// ```rust
+/// use rome_formatter::{format_tokens, format_token, IndentToken, LineToken, FormatToken, FormatOptions};
+/// let token = format_tokens!(
+///   "foo:",
+///   FormatToken::Space,
+///   "{",
+///   IndentToken::new(
+///     format_tokens!(FormatToken::Space, "bar:", FormatToken::Space, "lorem")
+///   ),
+///    FormatToken::Space,
+///   "}"
+/// );
+/// let unique_token = format_tokens!("single");
+/// let options = FormatOptions::default();
+/// assert_eq!(r#"foo: { bar: lorem }"#, format_token(&token, options).code());
+/// let options = FormatOptions::default();
+/// assert_eq!(r#"single"#, format_token(&unique_token, options).code());
```

Comment:
Nit: Split into two examples since barely anything except the use statement is reused
```suggestion
/// use rome_formatter::{format_tokens, format_token, IndentToken, LineToken, FormatToken, FormatOptions};
/// let token = format_tokens!(
///   "foo:",
///   FormatToken::Space,
///   "{",
///   IndentToken::new(
///     format_tokens!(FormatToken::Space, "bar:", FormatToken::Space, "lorem")
///   ),
///    FormatToken::Space,
///   "}"
/// );
/// assert_eq!(r#"foo: { bar: lorem }"#, format_token(&token, options).code());
/// ```
/// Or creating a single token
/// ```
/// use rome_formatter::{format_tokens, format_token, FormatOptions};
/// let options = ;
/// assert_eq!(r#"single"#, format_token(&format_tokens!("single"), FormatOptions::default()).code());
```